python/pyspark/sql/session.py

```diff
@@ -695,9 +704,10 @@ def prepare(obj):
             prepare = lambda obj: obj
```

Comment:
See https://github.com/apache/spark/blob/6f782efb044403ab3ca79662fdeca7f1f906e1bf/python/pyspark/sql/pandas/conversion.py#L312

@HyukjinKwon  That's why we need to do verification after the schema is inferred.

Reply:
1. create from pandas, schema is a list.
2. create directly without schema, schema is None

Reply:
Can we fix schema inference to infer the correct schema?

Reply:
it is not schema inference, it just extracts the name.

I think doing verification after the schema is inferred is good idea. Otherwise, we need to change when to infer schema.